### DISCUSSION TITLE
Add Favorite Application

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [mdgrs-mei/PowerShellRun#60] [Ideas]
I love now that I can create an entry group for Applications, but there are so many, some of which I never use. I would love a way to be able to select my favorite applications only. I haven't found a convenient way to do this. I suppose I could track down the link and add that as a favorite file,  but seems like a lot of work when you already can identify applications.

This would mean adding a command like `Add-PSRunFavoriteApplication` where I could specify the app name. It would be helpful if there was a command like `Get-PSRunApplication` that would list detected applications and their links.

**Comment by jdhitsolutions** [comment]
Digging into this topic further, I wonder if there is any benefit in separating Start Menu links from shell:AppsFolder apps.

**Comment by mdgrs-mei** [comment]
`Add-PSRunFavoriteApplication` sounds good to me. The point is how we can make the addition fast as searching for an app seems to be slow especially with shell::ApsFolder apps.

**Reply by jdhitsolutions** [reply]
I think the user needs to know the name or path in advance. That's why you need a command they can run to list what's available. I don't think performance is an issue here.

**Comment by jdhitsolutions** [comment]
I might be overthinking this. I can define a group and add apps as script blocks.

```powershell
$AppsGroup = Add-PSRunEntryGroup -Name ' Apps' -Description 'Desktop Apps' -Icon '🖥️' -PassThru
Add-PSRunScriptBlock -EntryGroup $AppsGroup -Name 'Open Vantage' -ScriptBlock { Start-Process  shell:AppsFolder\E046963F.LenovoSettingsforEnterprise_k1h2ywk1493x8!App ; psrun } -Description 'Open Lenovo Vantage' -Icon '🔋'
Add-PSRunScriptBlock -EntryGroup $AppsGroup -Name 'Spotify' -ScriptBlock { Start-Process C:\Users\Jeff\AppData\Roaming\Spotify\Spotify.exe ; psrun } -Description 'Open Spotify' -Icon '🎵'
Add-PSRunScriptBlock -EntryGroup $AppsGroup -Name 'Obsidian' -ScriptBlock { Start-Process shell:AppsFolder\md.obsidian} -Description 'Launch Obsidian' -Icon '📝'
```

![image](https://github.com/user-attachments/assets/3bf75840-e695-444d-9ecb-ad2005d734e8)
